crates/db_schema/src/impls/instance.rs

```diff
@@ -206,12 +213,72 @@ impl Instance {
   }
 }
 
-impl Blockable for InstanceActions {
-  type Form = InstanceBlockForm;
-  type ObjectIdType = InstanceId;
-  type ObjectType = Instance;
+impl InstanceActions {
+  pub async fn block_communities(
+    pool: &mut DbPool<'_>,
+    form: &InstanceCommunitiesBlockForm,
+  ) -> LemmyResult<Self> {
+    let conn = &mut get_conn(pool).await?;
+    insert_into(instance_actions::table)
+      .values(form)
+      .on_conflict((instance_actions::person_id, instance_actions::instance_id))
+      .do_update()
+      .set(form)
+      .returning(Self::as_select())
+      .get_result::<Self>(conn)
+      .await
+      .with_lemmy_type(LemmyErrorType::InstanceBlockCommunitiesAlreadyExists)
```

Comment:
The error name doesnt make sense as it uses upsert. In general it would be better if we rename all these error variants like `InstanceBlockCommunitiesError`, because there may be different reasons causing them.

Reply:
K, will fix.